src/files/routes.rs

```diff
@@ -49,7 +63,66 @@ async fn upload_file(
         }
     }
 
-    Ok((StatusCode::CREATED, Json(FileUploaded { id })))
+    Ok((StatusCode::CREATED, Json(FileWrapper { id })))
+}
+
+/// Axum service to handle file download requests
+/// FIXME: On integration, we need to do auth via token `Session` extrating on the request
+use axum_macros::debug_handler;
+#[debug_handler]
+async fn download_file(
+    // State(app): State<Arc<AppState>>,
+    // FIXME: harden extractor to only accept "*.filetype" or similar to prevent malicious client requests for "../../something.sh" and the like
+    Path(filename): Path<String>,
+    headers: HeaderMap,
+) -> Result<impl IntoApiResponse, AppError> {
+    let range = headers.get("range").ok_or(
+        AppError::new("Malformed file download request").with_status(StatusCode::NOT_ACCEPTABLE),
+    )?;
+
+    let filepath = format!("{}/{}", "/tmp", filename); // FIXME: use app state to set a file store location on disc
+    let mut file = match File::open(&filepath).await {
+        Ok(file) => file,
+        Err(_) => return Err(AppError::new("File not found").with_status(StatusCode::NOT_FOUND)),
+    };
+
+    let file_size = file.metadata().await.unwrap().len();
+    let range_str = range.to_str().unwrap();
+    // TODO: better way to ensure GET headers are correct and
+    if let Some(range) = range_str.strip_prefix("bytes=") {
+        let (start, end) = range.split_once('-').unwrap();
+        let start: u64 = start.parse().unwrap();
+        let end: u64 = end.parse().unwrap_or(file_size - 1);
+
+        file.seek(std::io::SeekFrom::Start(start)).await.unwrap();
```

Comment:
We seem to be trusting the user input on the start and end. it makes me wonder if this requires bounds checking. For example:
- what is the start is larger than the file?
- what if the end if larger than the file?